src/prefect/engine/result/base.py

```diff
@@ -21,12 +21,14 @@
 is `None`.
 """
 import copy
+import types
 import pendulum
 import uuid
-from typing import Any
+from typing import Any, Callable
 
 from prefect.engine.serializers import PickleSerializer, Serializer
 from prefect.utilities import logging
+from hashlib import md5
```

Comment:
Since md5 suffers from known hash collision attacks, perhaps we should consider sha256 or a different algorithm instead? I'm unsure what the support looks like, particularly for FIPS-compliant Python implementations, though.

Reply:
Md5 hash collisions require a carefully crafted input to achieve which I considered very unlikely to happen I also choose it because its somewhat fast and short. But surely if it would cause a problem it has to change